cocotb/share/lib/simulator/simulatormodule.cpp

```diff
@@ -270,6 +270,24 @@ static PyObject *log_msg(PyObject *self, PyObject *args)
 }
 
 
+static p_callback_data make_callback_data(PyObject *func, PyObject *args, PyObject *kwargs)
```

Comment:
```suggestion
static p_callback_data p_callback_data_new(PyObject *func, PyObject *args, PyObject *kwargs)
```

to indicate it's an allocating function for the p_callback_data type.

Reply:
Should that be `s_callback_data_new`, since `p_callback_data` is `s_callback_data*`?

Reply:
Would make sense, yes. That being said, the naming of this struct and the two associated typedefs look quite unusual to me. I'd go for a normal (yet not fully clean naming-wise) `callback_data_t` or (my preferred option) simply using `struct callback_data`.

Reply:
(The struct naming is obviously not part of this PR.)

Reply:
I've gone ahead and added an extra commit to rename `s_callback_data` to `callback_data`. This is C++ code now, so `struct callback_data` and `callback_data` are synonymous anyway.